plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -164,27 +166,30 @@ def list_all_project_variables(self):
             vars_page = self.project.variables.list(page=page_nb)
         return variables
 
-    def create_variable(self, key, value, masked, protected, variable_type):
+    def create_variable(self, key, value, masked, protected, variable_type, environment_scope):
         if self._module.check_mode:
             return
         return self.project.variables.create({"key": key, "value": value,
                                               "masked": masked, "protected": protected,
-                                              "variable_type": variable_type})
+                                              "variable_type": variable_type,
+                                              "environment_scope": environment_scope})
```

Comment:
Is `environment_scope` supported by all versions of GitLab / the library this module supports? If not, it's probably a good idea to only provide this if it is not empty.

Reply:
python-gitlab version 2.5.0 [https://github.com/python-gitlab/python-gitlab/commit/4492fc42c9f6e0031dd3f3c6c99e4c58d4f472ff]

Reply:
I guess in that case it's better to pass it to `create()` if it's non-empty.

Reply:
```
        var = {
            "key": key, "value": value,
            "masked": masked, "protected": protected,
            "variable_type": variable_type
        }
        if environment_scope is not None:
            var["environment_scope"] = environment_scope
        return self.project.variables.create(var)
```
like that @felixfontein ?

Reply:
Yes, except that it probably never is explicitly `None`, since you default to `'*'` in the caller.

Are `None` and `'*'` equivalent?

Reply:
Are not equivalent. the scope by default is '*'. 
I will remove the default value to detect that it is defined.

Reply:
@castorinop any updates?

Reply:
@castorinop Are you working on this? Let us know.